tools/go.mod

```diff
@@ -1,6 +1,8 @@
 module tools
 
-go 1.23.7
+go 1.24.0
+
+toolchain go1.24.4
```

Comment:
Similar comment to https://github.com/hashicorp/terraform-plugin-go/pull/556#discussion_r2337598276, but not a blocker 👍🏻